### ISSUE TITLE
method "PUT" sending empty response in edit.blade.php on submit .

### ISSUE THREAD

**Issue by sahnb** [protonemedia/laravel-splade#69]
![image](https://user-images.githubusercontent.com/8742640/194803100-4fe5134f-5021-48d3-b38f-a6908d270723.png)

![image](https://user-images.githubusercontent.com/8742640/194803163-9e43f747-65a3-4a5c-b621-d655fa187342.png)

![image](https://user-images.githubusercontent.com/8742640/194803212-7c4cbdd5-b035-455e-bf1b-e8b36a1c6ebd.png)
![image](https://user-images.githubusercontent.com/8742640/194803236-1a49b93e-2a2d-4b4f-a8f0-ebc0498b8a87.png)

**Comment by sahnb** [comment]
need to define update route separately.

**
![image](https://user-images.githubusercontent.com/8742640/194818938-a6bf6066-48f0-4006-b5aa-02b98b82d511.png)
**

**Comment by Jubeki** [comment]
@sahnb About the problem you are having:

```blade
<x-splade-form action="..." method="put">...</x-splade-form>
```

This does translate to:
```html
<form action="..." method="PUT">...</form>
```

but this is incorrect, because the method can only be `GET` or `POST`
you would instead need to do the following:

```blade
<x-splade-form action="..." method="POST">
    @method('PUT')
    ...
</x-splade-form>
```

Which translates to
```html
<form action="..." method="POST">
    <input type="hidden" name="_method" value="PUT" />
    ...
</form>
```

See https://stackoverflow.com/questions/8054165/using-put-method-in-html-form

**Comment by pascalbaljet** [comment]
@Jubeki Splade's Form component does that for you. You don't have to manually use the `@method` directive or a hidden input field.

**Comment by Sh1d0w** [comment]
I am getting the same issue @pascalbaljet can you give an example?

I have the exact same markup as the OP, using `<x-splade-form .. method="PUT">` and then the request body is just empty when I dump it in the controller, as HTML forms support only GET and POST methods.

Any advice would be appreciated.

**Comment by Sh1d0w** [comment]
Solved it by adding

```
@if(isset($data['id']))
        <x-splade-input name="id" type="hidden"/>
        <x-splade-input name="_method" type="hidden"/>
    @endif
```

And providing `['_method' => 'PUT', .... ]` along with the data.

**Comment by pascalbaljet** [comment]
Whoops, I made a wrong assumption. It's fixed in v0.6.1 and I also added a test for it :)

**Comment by J87NL** [comment]
At first, thanks Pascal for your amazing work, Splade is exactly what is was looking for for years!

Unfortunately an issue related to described above seems to be still there for PUT/PATCH requests, at least in my tests.

I'm using Splade 0.6.13 (with protonemedia/laravel-splade-breeze) / Laravel 9.38.0 / PHP 8.0.22

POST-requests for creating are working just fine, but when I do an PUT or PATCH request (with `<x-splade-form method="patch" action="...`)  the post-data is empty. In the network-tab I do see there was a PUT/PATCH-request.

When I remove the `method="patch"` and add an extra route like `Route::post('/test/{item}', [...Controller::class, 'update'])->name('test');` and use this for the action, the POST-data is there.

**Comment by J87NL** [comment]
My apologies, apparently my package.json was running behind having "@protonemedia/laravel-splade": "^0.5.0" instead of "0.6.0". I did update with composer but not the npm-packages.

I installed a fresh Laravel application with the latest protonemedia/laravel-splade-breeze, placed my models, views and controllers in it and now it's working like a charm.

This fixes https://github.com/protonemedia/laravel-splade/issues/102 as well.
